tests/security/ibmtss/ibmtss_basic_function.pm

```diff
@@ -0,0 +1,62 @@
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Summary: Update IBM's Trusted Computing Group Software Stack (TSS) to the latest version.
+#          IBM has tested x86_64, s390x and ppc64le, we only need cover aarch64
+#          This test module covers basic function test
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#101088, poo#102792, tc#1769800
+
+use base 'opensusebasetest';
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils 'zypper_call';
+use version_utils 'is_sle';
+use registration 'add_suseconnect_product';
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    # Install required develop packages
+    if (is_sle('>=15')) {
+        add_suseconnect_product('sle-module-desktop-applications');
+        add_suseconnect_product('sle-module-development-tools');
+    }
+    zypper_call('in -t pattern devel_basis');
+    zypper_call('in openssl-devel');
+
+    # Install emulated tpm server
+    zypper_call('in ibmswtpm2');
+
+    # Swith to root console to start tpm server in backaround
+    select_console('root-console');
+    if (is_sle) {
+        background_script_run('/usr/lib/ibmtss/tpm_server');
+    }
+    else {
+        background_script_run('/usr/libexec/ibmtss/tpm_server');
+    }
+
+    # Download the test script, which is imported from link 'https://git.code.sf.net/p/ibmtpm20tss/tssi'
+    $self->select_serial_terminal;
+
+    assert_script_run('git clone https://git.code.sf.net/p/ibmtpm20tss/tss ibmtpm20tss-tss', timeout => 240);
+    assert_script_run('cd ibmtpm20tss-tss/utils');
+    assert_script_run('make -f makefiletpmc', timeout => 120);
+
+    # Modify the script to use the binaries installed in current system
+    assert_script_run q(sed -i 's/^PREFIX=.*/PREFIX=tss/g' reg.sh);
+
+    # Run the script
+    assert_script_run('export TPM_INTERFACE_TYPE=socsim');
+    script_run('./reg.sh | tee /tmp/tss.log', timeout => 120);
+    upload_logs('/tmp/tss.log');
+
+    # Check the test result
+    assert_script_run q(cat /tmp/tss.log | grep 'Success - 0 Tests 0 Warnings');
```

Comment:
Please use a variable to define /tmp/tss.log for tidy code.

Reply:
thanks, will do